test/test_hlo_metadata.py

```diff
@@ -0,0 +1,96 @@
+# Parse local options first, and rewrite the sys.argv[].
+# We need to do that before import "common", as otherwise we get an error for
+# unrecognized arguments.
+import argparse
+import sys
+
+parser = argparse.ArgumentParser(add_help=False)
+parser.add_argument('--replicated', action='store_true')
+parser.add_argument('--long_test', action='store_true')
+parser.add_argument('--max_diff_count', type=int, default=25)
+parser.add_argument('--verbosity', type=int, default=0)
+FLAGS, leftovers = parser.parse_known_args()
+sys.argv = [sys.argv[0]] + leftovers
+
+# Normal imports section starts here.
+import torch
+import torch_xla
+import torch_xla.utils.utils as xu
+import torch_xla.core.xla_model as xm
+import torch_xla.debug.metrics as met
+import unittest
+import json
+
+
+class TestHloMetaData(unittest.TestCase):
+
+  def setUp(self):
+    super(TestHloMetaData, self).setUp()
+
+  def tearDown(self):
+    super(TestHloMetaData, self).tearDown()
+
+  def test_metadata(self):
+    torch_xla._XLAC._set_ir_debug(True)
+    layer1 = torch.nn.Linear(4, 4)
+    nl1 = torch.nn.ReLU()
+    layer2 = torch.nn.Linear(4, 2)
+    nl2 = torch.nn.Tanh()
+    model = torch.nn.Sequential(layer1, nl1, layer2, nl2)
+    model = model.to(device=xm.xla_device())
+    inp = torch.rand(4, 4, device=xm.xla_device())
+    out = model(inp)
+    ctx = torch_xla._XLAC.lowering.LoweringContext()
+    ctx.build([out])
+    hlo_text = ctx.hlo_json()
+
+    # Strings to match in the lowering
+    bingo = {
+        "torch/nn/modules/linear.py": False,
+        "torch/nn/modules/activation.py": False,
+        "torch/nn/functional.py": False,
+        "Sequential[model]/Linear[0]": False,
+        "Sequential[model]/ReLU[1]": False,
+        "Sequential[model]/Linear[2]": False,
+        "Sequential[model]/Tanh[3]": False,
+        "aten__addmm": False,
+        "aten__relu": False,
+        "aten__tanh": False,
+        "aten__permute": False
+    }
+
+    non_zero_metadata = False
+
+    local_json = json.loads(hlo_text)
+    assert "computations" in local_json
+    for c in local_json["computations"]:
+      if "instructions" in c:
+        i = c["instructions"]
+        for op in i:
+          if 'metadata' in op:
+            meta = op["metadata"]
+            print(meta)
+            if len(meta) > 0:
+              non_zero_metadata = True
+            for km, vm in meta.items():
+              for k in bingo.keys():
+                if isinstance(vm, str) and k in vm:
+                  bingo[k] = True
+
+    assert non_zero_metadata, "No metadata was lowered - an issue with turning on IR DEBUG?"
+
+    for k, v in bingo.items():
+      assert v, f"Keyword {k} was not found as expected in HLO metadata for simple test"
+
+    print("All required metadata symbols matched")
+
+
+if __name__ == '__main__':
+  torch.set_default_tensor_type('torch.FloatTensor')
```

Comment:
Can these logic be moved to a `setUp` or `setUpClass` method of the test class?

Reply:
This was pretty much copy and paste from an existing test - I can refactor